tiledb/sm/query/result_tile.cc

```diff
@@ -59,7 +59,7 @@ ResultTile::ResultTile(
   coord_tiles_.resize(domain_->dim_num());
   attr_tiles_.resize(array_schema->attribute_num());
   for (uint64_t i = 0; i < array_schema->attribute_num(); i++) {
-    const Attribute* attribute = array_schema->attribute(i);
+    tdb_shared_ptr<const Attribute> attribute = array_schema->attribute(i);
```

Comment:
Declaring these as `auto` is good too. As I've said above, we shouldn't be dropping `const` on this function declaration anyway. It doesn't look like it needs to change.